src/ufixed.jl

```diff
@@ -35,14 +31,7 @@ const uf14 = UFixedConstructor{UInt16,14}()
 const uf16 = UFixedConstructor{UInt16,16}()
 
 zero{T,f}(::Type{UFixed{T,f}}) = UFixed{T,f}(zero(T),0)
-for uf in UF
-    TT = rawtype(uf)
-    f = nbitsfrac(uf)
-    T = UFixed{TT,f}
-    @eval begin
-        one(::Type{$T}) = $T($(2^f-1),0)
-    end
-end
+one{T<:UFixed}(::Type{T}) = T((2^nbitsfrac(T)-1),0)
```

Comment:
This will be costly on v0.4. On v0.5 we can constant fold this, but only if we outline the `2^nbitsfrac(T)` into a pure function `Base.@pure one_pattern(f)=(2^f-1)`, but this won't work on v0.4.

One solution would be for now to use the current way on v0.4 with your method as a fallback, and on v0.5 and higher use pure functions on the type-parameter to get the same effect.